specs/PrintToPdf.md

```diff
@@ -0,0 +1,303 @@
+
+# API Notes
+
+
+# API Details
+```
+interface ICoreWebView2_4;
+interface ICoreWebView2Environment5;
+interface ICoreWebView2PrintSettings;
+interface ICoreWebView2PrintToPdfCompletedHandler;
+
+
+/// The orientation for printing, used by the `Orientation` property on
+/// `ICoreWebView2SPrintSettings`. Currently only printing to PDF
+/// is supported.
+[v1_enum]
+typedef enum COREWEBVIEW2_PRINT_ORIENTATION
+{
+  /// Print the page(s) in portrait orientation.
+  COREWEBVIEW2_PRINT_ORIENTATION_PORTRAIT,
+
+  /// Print the page(s) in landscape orientation.
+  COREWEBVIEW2_PRINT_ORIENTATION_LANDSCAPE,
+} COREWEBVIEW2_PRINT_ORIENTATION;
+
+[uuid(9eae81c3-fe02-4084-b475-903a4ed9252e), object, pointer_default(unique)]
+interface ICoreWebView2_4 : IUnknown {
+  /// Print the current page to PDF asynchronously with the provided settings.
+  /// See `ICoreWebView2PrintSettings` for description of settings.
+  ///
+  /// Use `resultFilePath` to specify the path to the PDF file. The host should
+  /// provide an absolute path, including file name. If the path
+  /// points to an existing file, the file will be overwritten. If
+  /// `resultFilePath` is invalid, `isSuccessful` in
+  /// `ICoreWebView2PrintToPdfCompletedHandler` is set to `FALSE` and the
+  /// returned `resultFilePath` is an empty string.
+  ///
+  /// The async `PrintToPdf` operation completes when the data has been written
+  /// to the PDF file. At this time the `ICoreWebView2PrintToPdfCompletedHandler`
+  /// is invoked.
+  ///
+  /// \snippet FileComponent.cpp PrintToPdf
+  HRESULT PrintToPdf(
+    [in] LPCWSTR resultFilePath,
+    [in] ICoreWebView2PrintSettings* printSettings,
+    [in] ICoreWebView2PrintToPdfCompletedHandler* handler);
+}
+
+/// Receives the result of the `PrintToPdf` method. The `resultFilePath`
+/// contains the final path of the PDF file.
+[uuid(4808ac58-c372-4d3d-be5e-900df2593835), object, pointer_default(unique)]
+interface ICoreWebView2PrintToPdfCompletedHandler : IUnknown {
+
+  /// Provides the result of the corresponding asynchronous method.
+  HRESULT Invoke([in] HRESULT errorCode, [in] LPCWSTR resultFilePath);
+}
+
+/// Settings used by the `PrintToPdf` method. Other programmatic printing is not
+/// currently supported.
+[uuid(30964a64-2e92-4604-9eaa-3afcd3a015dc), object, pointer_default(unique)]
+interface ICoreWebView2PrintSettings : IUnknown {
+
+  /// The orientation can be portrait or landscape. The default orientation is
+  /// portrait. See `COREWEBVIEW2_PRINT_ORIENTATION`.
+  [propget] HRESULT Orientation(
+    [out, retval] COREWEBVIEW2_PRINT_ORIENTATION* orientation);
+
+  /// Sets the `Orientation` property.
+  [propput] HRESULT Orientation(
+      [in] COREWEBVIEW2_PRINT_ORIENTATION orientation);
+
+  /// The scale factor is a value between 0.1-2.0. The default is 1.0.
+  [propget] HRESULT ScaleFactor([out, retval] double* scaleFactor);
+
+  /// Sets the `ScaleFactor` property. In an invalid value is provided, the
+  /// default is used.
```

Comment:
For all these properties where you say the default is used - does that mean if I try to set an invalid value instead it will be set to the default? Or does it mean the set to an invalid value will be ignored and there's no change to the value of the property?

If I do the following:
```
e.ScaleFactor = 2.0;
e.ScaleFactor = 0;
e.ScaleFactor; // is it 1.0 or 2.0 now?
```